homeassistant/components/nederlandse_spoorwegen/config_flow.py

```diff
@@ -0,0 +1,141 @@
+"""Adds config flow for Nederlandse Spoorwegen."""
+
+import asyncio
+from typing import Any
+
+import ns_api
+from ns_api import RequestParametersError
+import voluptuous as vol
+
+from homeassistant.config_entries import (
+    ConfigEntry,
+    ConfigFlow,
+    ConfigFlowResult,
+    ConfigSubentryFlow,
+    SubentryFlowResult,
+)
+from homeassistant.const import CONF_API_KEY, CONF_NAME
+from homeassistant.core import callback
+from homeassistant.helpers.selector import TimeSelector
+
+from .const import (
+    CONF_STATION_FROM,
+    CONF_STATION_TO,
+    CONF_STATION_VIA,
+    CONF_TIME,
+    DOMAIN,
+)
+
+
+async def get_stations(nsapi: ns_api.NSAPI):
+    """Return all available stations."""
+    loop = asyncio.get_running_loop()
+    return await loop.run_in_executor(None, nsapi.get_stations)
+
+
+class NederlandseSpoorwegenConfigFlow(ConfigFlow, domain=DOMAIN):
+    """Nederlandse Spoorwegen config flow."""
+
+    # VERSION = 1
+    # MINOR_VERSION = 1
+
+    def __init__(self) -> None:
+        """Initialize the config flow."""
+
+        self.api_key: str
+
+    @classmethod
+    @callback
+    def async_get_supported_subentry_types(
+        cls, config_entry: ConfigEntry
+    ) -> dict[str, type[ConfigSubentryFlow]]:
+        """Return subentries supported by this integration."""
+        return {"trip": NederlandseSpoorwegenSubentryFlowHandler}
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> ConfigFlowResult:
+        """Handle a flow initialized by the user."""
+        errors: dict[str, str] = {}
+        if user_input is not None:
+            self.api_key = user_input[CONF_API_KEY]
+            try:
+                await get_stations(ns_api.NSAPI(self.api_key))
+            except RequestParametersError:
+                errors["base"] = "invalid_api_key"
+            else:
+                await self.async_set_unique_id(self.api_key)
```

Comment:
the api key is not a good unique id. Do we have something to match this to a user id? Otherwise we just need to use `async_abort_match_entries_match` to match on the data